### PR TITLE
ocamlPackages.ounit(2): 2.2.2 → 2.2.3

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

<https://github.com/gildor478/ounit/releases/tag/v2.2.3>

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
@GrahamcOfBorg build ocamlPackages.batteries ocamlPackages.irmin-unix

File: pkgs/development/ocaml-modules/ounit2/default.nix [hunk 1]
```diff
@@ -1,14 +1,14 @@
 { lib, buildDunePackage, fetchurl, stdlib-shims }:
 
 buildDunePackage rec {
-  minimumOCamlVersion = "4.02.3";
+  minimumOCamlVersion = "4.04";
```
Comment (vbgl) [inline pkgs/development/ocaml-modules/ounit2/default.nix:4]: This change “removes” many packages: nixpkgs-review lists 92 of them. Is it an issue?

Comment (sternenseemann) [reply pkgs/development/ocaml-modules/ounit2/default.nix:4]: `nixpkgs-review` interestingly didn't list any for me, did you use any special options or do `rev` and `pr` behave differently?

If you think, it's worth it to fix 4.02 and 4.04 OCaml packages that depend on ounit, I'll go through them and add a

```nix
  doCheck = lib.versionAtLeast ocaml.version ounit.minimumOCamlVersion;
```

I myself really am not sure, it seems kinda sensible, but I guess if we do this for every such increase of `minimumOCamlVersion` this becomes a huge task?!

**Comment by vbgl** [comment]
@GrahamcOfBorg build beluga flow fstar jackline ocamlformat ocaml-top patdiff satysfi vagrant virt-top

**Pull request merged**